### DISCUSSION TITLE
Any support for surrealdb planned?

### DISCUSSION THREAD

**Discussion by Joey92** [theory/sqitch#782] [Ideas]
As the title says. I've been eying SurrealDB lately. Currently coming from Postgres, I'd be totally missing sqitch.

**Comment by theory** [comment]
First I've heard of it, so I can't say how much work it'd be, but offhand it looks like they only client library currently available is in JavaScript (from the [SDKs page](https://surrealdb.com/docs/integration/sdks)). Maybe there is ODBC support? Other than that, if it has robust SQL functionality it ought not bee too difficult for an interested party to take on. I'd be happy to advise on the effort and test it!

**Comment by Joey92** [comment]
I took a look at the currently supported databases. I can see that JDBC or ODBC is preferred? SurrealDB seems to use HTTP as a first class citizen with either plain REST or WebSockets. What's yuor opinions on that?

**Reply by theory** [reply]
Each engine can use whatever interface it deems best, but if you can use SQL with the DBI there likely will be a lot less code to write. Snowflake's main API is also REST, but the company provides an ODBC driver and CLI that Sqitch uses.

I'm assuming that Surreal provides an CLI, yes? That's a key feature of Sqitch: It uses the native CLI SQL file format for each engine.
